packages/schema/src/Schema.ts

```diff
@@ -1819,16 +1819,30 @@ export declare namespace Struct {
   /**
    * @since 1.0.0
    */
-  export type Type<F extends Fields> =
-    & { readonly [K in Exclude<keyof F, TypeTokenKeys<F>>]: Schema.Type<F[K]> }
-    & { readonly [K in TypeTokenKeys<F>]?: Schema.Type<F[K]> }
```

Comment:
are those 2 just selecting/filtering keys from `F`? could they be replaced with `as`?

Reply:
Ill put up a PR soon with a different change in an attempt to address this

Reply:
You mean https://github.com/Effect-TS/effect/pull/2435 ?

Reply:
Yeah, exactly. This should work fine here and is the most straightforward for the compiler to handle. I see that u had to had to suppress one error but that's likely fine (it's in an internal implementation anyway. I could investigate why this fails at some other time.

Reply:
While I agree it would make it lose order of keys, not that it matters too much

Reply:
Would be curious to know why it fails indeed

Reply:
order of keys? what do u mean?

Reply:
it would put all optional keys at the end of the object, so `{ a: S.number, b: S.optional(S.number), c: S.number }` would become in type: `{ a: number; c: number; b?: number }` instead of `{ a: number; b?: number; c: number }`

Reply:
This was already the case though, but I kinda do like that the new solution fixes that

Reply:
I remember this was the reason I didn't go for `as` in the old schema

Reply:
I guess I just wouldn't be bothered with the key order here ;p You could try to preserve the order using something like this: [TS playground](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true#code/PTAEAEFMA8EMGMAuB5ADoglgewHawDYAKATlqpMYgJ4AqV5AzgFyiLECukAUNeaAEqQAjuwzFIAExJkGoALygA5ACNFXHvUigAIlgDKidgDMjAdQyIAFgeNGAPDQB880AG8uoUAG0A0qAw4oADWkFRYRqA0ALoA-CxeNL5RUQDcoCCslhiyAO5Y7PgSoJawAG5aiFigkPgYALYBsIhaAAYAPqDsOBKQRgGSLaBGpHWZWljKiLD9RbzcAL6gAGRuHulgYeyg8LCB8Fh1qOzNY6C1DIig4acAROIiYpI3oKikqLISVHgNO-j4VGtfP5AoIHuIpG8GFEWF0gjgsDkcCkuPNkRo+ABZKg2EwudyeYAAKkJoDyOVAhOAa1gLBw7DqygoyIJxNAdXY8EsFKpnmULAuxACAHNmelWbAcpAGActJS1vAWMosFh8JBdsjUeo5gIpQVLgpdDizBZrIYTHYsUbHGjtYIGHrpKgXHa9V4bsobqktZpQABhA5HZoOZwKVzePwBYKha7RFiJHxRUCLFaw+GIm0+gBijXwLvw+r9AeOkDshrNxqsRot2PLjmt6gyFmKsFkuC0MDYsG2ZTVl0sFAqlnyQssABpQOitNm8EQ3i5pwQ84g3R6vUA)